packages/react-native-reanimated/Common/cpp/Fabric/ReanimatedCommitHook.cpp

```diff
@@ -38,23 +40,21 @@ RootShadowNode::Unshared ReanimatedCommitHook::shadowTreeWillCommit(
   // ShadowTree not commited by Reanimated, apply updates from PropsRegistry
 
   auto rootNode = newRootShadowNode->ShadowNode::clone(ShadowNodeFragment{});
+  std::unordered_map<
+      const ShadowNodeFamily *,
+      std::vector<std::shared_ptr<RawProps>>>
+      propsMap;
 
   {
     auto lock = propsRegistry_->createLock();
 
     propsRegistry_->for_each(
         [&](const ShadowNodeFamily &family, const folly::dynamic &props) {
-          auto newRootNode =
-              cloneShadowTreeWithNewProps(rootNode, family, RawProps(props));
-
-          if (newRootNode == nullptr) {
-            // this happens when React removed the component but Reanimated
-            // still tries to animate it, let's skip update for this specific
-            // component
-            return;
-          }
-          rootNode = newRootNode;
+          propsMap[&family].push_back(std::make_shared<RawProps>(props));
```

Comment:
Let's use `emplace_back` instead if possible.

Reply:
out of curiosity, what's the difference?

Reply:
`emplace_back` uses move semantics

Reply:
ooh, didn't know 😄